### PR TITLE
 feat(Chiasenhac): add presence

### PR DESCRIPTION
**Pull request by kyrie25**
**Add presence for [Chiasenhac](https://chiasenhac.vn/)** (region restricted to Vietnam and a small number of countries)

<details>
<summary> Proof of the presnce working: </summary>
<br>

image
image
image
image
image
image

</details>

### CODE REVIEW

File: websites/C/Chiasenhac/dist/metadata.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
+{
+  "$schema": "https://schemas.premid.app/metadata/1.6",
+  "author": {
+    "name": "Kyrie",
+    "id": "368399721494216706"
+  },
+  "service": "Chiasenhac",
```
Comment (Bas950) [inline websites/C/Chiasenhac/dist/metadata.json:7]: is that an official version of the name? Otherwise just use the version in altnames

**Review by Bas950 — DISMISSED** [review]

File: websites/C/Chiasenhac/dist/metadata.json (hunk 1 above)
Comment (Bas950) [reply websites/C/Chiasenhac/dist/metadata.json:7]: tbh I would still put the original/primary name and then the other as altname. I am guessing the 2 should be switched around.

Comment (kyrie25) [reply websites/C/Chiasenhac/dist/metadata.json:7]: Will change ASAP

File: websites/C/Chia Sẻ Nhạc/presence.ts [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      presenceData.details = document.querySelector<HTMLHeadingElement>(
+        "body > section > div.container > div > div.col-md-9 > div:nth-child(4) > div.col-md-4 > div > div.card-body > h2"
+      ).textContent;
+      presenceData.state = document
+        .querySelector<HTMLLIElement>(
+          "body > section > div.container > div > div.col-md-9 > div:nth-child(4) > div.col-md-4 > div > div.card-body > ul > li:nth-child(1)"
+        )
+        .textContent.slice(7);
+      if (cover) {
+        delete presenceData.largeImageKey;
```
Comment (EncryptedDev) [inline websites/C/Chia Sẻ Nhạc/presence.ts:52]: You do not need to delete it, the original value is automatically replaced

File: websites/C/Chia Sẻ Nhạc/presence.ts [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        presenceData.largeImageKey = document.querySelector<HTMLImageElement>(
+          "#companion_cover > img"
+        ).src;
+      }
+      if (timestamps) {
+        delete presenceData.startTimestamp;
```
Comment (EncryptedDev) [inline websites/C/Chia Sẻ Nhạc/presence.ts:58]: Here too

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/C/Chia Sẻ Nhạc/presence.ts (hunk 2 above)
Comment (kyrie25) [reply websites/C/Chia Sẻ Nhạc/presence.ts:52]: Oh interesting, I'll fix it ASAP

Comment (kyrie25) [reply websites/C/Chia Sẻ Nhạc/presence.ts:52]: Done

Comment (Slowlife01) [reply websites/C/Chia Sẻ Nhạc/presence.ts:52]: Still here

Comment (kyrie25) [reply websites/C/Chia Sẻ Nhạc/presence.ts:52]: Brainfart 💀 
Done fr this time

**Review by Slowlife01 — APPROVED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**